data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilter.java

```diff
@@ -298,17 +298,32 @@ public void testNoNulls() {
 
   @Test
   public void testIsNaN() {
+    Assume.assumeTrue("Avro files do not have row group statistics", format != FileFormat.AVRO);
+
     boolean shouldRead = shouldRead(isNaN("all_nans"));
-    Assert.assertTrue("Should read: NaN counts are not tracked in Parquet metrics", shouldRead);
+    Assert.assertTrue(String.format("Should read: NaN counts are not tracked in %s metrics", format), shouldRead);
 
     shouldRead = shouldRead(isNaN("some_nans"));
-    Assert.assertTrue("Should read: NaN counts are not tracked in Parquet metrics", shouldRead);
-
-    shouldRead = shouldRead(isNaN("no_nans"));
-    Assert.assertTrue("Should read: NaN counts are not tracked in Parquet metrics", shouldRead);
+    Assert.assertTrue(String.format("Should read: NaN counts are not tracked in %s metrics", format), shouldRead);
 
     shouldRead = shouldRead(isNaN("all_nulls"));
     Assert.assertFalse("Should skip: all null column will not contain nan value", shouldRead);
+
+    if (format == FileFormat.ORC) {
+      testIsNanOrc();
+    } else {
+      testIsNanParquet();
+    }
+  }
+
+  private void testIsNanParquet() {
+    boolean shouldRead = shouldRead(isNaN("no_nans"));
+    Assert.assertTrue("Should read: NaN counts are not tracked in Parquet metrics", shouldRead);
+  }
+
+  private void testIsNanOrc() {
+    boolean shouldRead = shouldRead(isNaN("no_nans"));
+    Assert.assertFalse("Should skip: no nans will not contain nan value", shouldRead);
```

Comment:
Good work @hililiwei! I like this a lot more.

I think we might be ok without the `testIsNan${format}` helper methods. We can just inline within the test as most of the logic is shared.

So something like the following, now that I see that so much of the logic can be shared.
```java
shouldRead = shouldRead(isNaN("no_nans"));
if (format == FileFormat.ORC) {
  Assert.assertFalse("Should skip: no nans will not contain nan value", shouldRead);
} else {
  Assert.assertTrue("Should read: NaN counts are not tracked in Parquet metrics", shouldRead);
}
```

Reply:
For the ORC assertion message, it might be more clear if we're a little more descriptive, given that we've stated that the `NaN counts are not tricked in orc metrics` already.

Something like `Should skip: The existence of NaN in a column is tracked in ORC metrics`. WDYT?

Reply:
> Good work @hililiwei! I like this a lot more.
> 
> I think we might be ok without the `testIsNan${format}` helper methods. We can just inline within the test as most of the logic is shared.
> 

good!